tests/testthat/test-add.R

```diff
@@ -164,6 +164,21 @@ test_that("can recycle when adding columns", {
   expect_identical(df_new$c, 3:1)
 })
 
+test_that("can recycle when adding a column of length 1", {
+  df <- tibble(a = 1:3)
+  df_new <- add_column(df, b = 4)
+  expect_identical(ncol(df_new), ncol(df) + 1L)
+  expect_identical(df_new$b, rep(4, 3))
+})
```

Comment:
No particular reason. My test is weaker, too -- doesn't test order of columns. Feel free to adapt.